modules/dokkatoo-plugin/src/main/kotlin/formats/DokkatooFormatTasks.kt

```diff
@@ -43,63 +44,56 @@ class DokkatooFormatTasks(
       offlineMode.convention(publication.offlineMode)
       pluginsConfiguration.addAllLater(providers.provider { publication.pluginsConfiguration })
       pluginsClasspath.from(
-        dependencyContainers.dokkaPluginsIntransitiveClasspath.map { classpath ->
-          classpath.incoming.artifacts.artifactFiles
-        }
+        depsManager.dokkaPluginsIntransitiveClasspathResolver
       )
       suppressInheritedMembers.convention(publication.suppressInheritedMembers)
       suppressObviousFunctions.convention(publication.suppressObviousFunctions)
     }
   }
 
-  val generatePublication = project.tasks.register<DokkatooGenerateTask>(
-    taskNames.generatePublication,
-    publication.pluginsConfiguration,
-  ).configuring task@{
-    description = "Executes the Dokka Generator, generating the $formatName publication"
-    generationType.set(DokkatooGenerateTask.GenerationType.PUBLICATION)
+  val generatePublication: TaskProvider<DokkatooGenerateTask> =
+    project.tasks.register<DokkatooGenerateTask>(
+      taskNames.generatePublication,
+      publication.pluginsConfiguration,
+    ).configuring task@{
+      description = "Executes the Dokka Generator, generating the $formatName publication"
+      generationType.set(DokkatooGenerateTask.GenerationType.PUBLICATION)
 
-    outputDirectory.convention(dokkatooExtension.dokkatooPublicationDirectory.dir(formatName))
+      outputDirectory.convention(dokkatooExtension.dokkatooPublicationDirectory.dir(formatName))
 
-    generator.apply {
-      // depend on Dokka Module Descriptors from other subprojects
-      dokkaModuleFiles.from(
-        dependencyContainers.dokkaModuleConsumer.map { modules ->
-          modules.incoming
-            .artifactView { componentFilter(LocalProjectOnlyFilter) }
-            .artifacts.artifactFiles
-        }
-      )
-    }
+      generator.apply {
+        this.moduleDescriptors.addAllLater(providers.provider {
+          this@DokkatooFormatTasks.moduleDescriptors
+        })
+      }
 
-    applyFormatSpecificConfiguration()
-  }
+      // ugly hack, workaround for https://github.com/gradle/gradle/issues/13590
+      dependsOn(providers.provider {
+        this@DokkatooFormatTasks.moduleDescriptors.map { it.moduleGenerateTaskPath }
+      })
```

Comment:
And here's the end result of the ugly hack. The task paths of the module-generation tasks are extracted from the incoming Artifact Attributes and passed in as a dependency here.